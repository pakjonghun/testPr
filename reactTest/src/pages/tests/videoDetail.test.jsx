import { Route } from 'react-router-dom';
import { withRoute } from '../../tests/utils';
import VideoDetail from '../VideoDetail';
import { videos } from '../../tests/videos';
import { screen, render } from '@testing-library/react';
import ChannelInfo from '../../components/ChannelInfo';
import RelatedVideos from '../../components/RelatedVideos';

jest.mock('../../components/ChannelInfo.jsx');
jest.mock('../../components/RelatedVideos.jsx');

describe('videoDetail', () => {
  beforeEach(() => {
    ChannelInfo.mockReset();
    RelatedVideos.mockReset();
  });

  it('should display state data', () => {
    const { asFragment } = render(
      withRoute(<Route path="/" element={<VideoDetail />} />, [
        {
          state: { video: videos[0] },
          path: '/',
          key: 'fake-key',
        },
      ])
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('스테이트와 데이터가 일치해야 한다.', () => {
    render(
      withRoute(<Route path="/" element={<VideoDetail />} />, [
        {
          state: { video: videos[0] },
          path: '/',
          key: 'fake-key',
        },
      ])
    );

    const { title, description, channelId, channelTitle } = videos[0].snippet;
    screen.getByText(title);
    screen.getByText(description);
    expect(ChannelInfo).toHaveBeenCalledTimes(1);
    expect(RelatedVideos).toBeCalledTimes(1);
    expect(ChannelInfo.mock.calls[0][0]).toStrictEqual({
      id: channelId,
      name: channelTitle,
    });
    expect(RelatedVideos.mock.calls[0][0]).toStrictEqual({
      id: videos[0].id,
    });
  });
});
