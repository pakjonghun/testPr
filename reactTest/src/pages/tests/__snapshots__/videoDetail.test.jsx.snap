// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`videoDetail should display state data 1`] = `
<DocumentFragment>
  <section
    class="flex flex-col lg:flex-row"
  >
    <article
      class="basis-4/6"
    >
      <iframe
        frameborder="0"
        height="640"
        id="player"
        src="https://www.youtube.com/embed/1"
        title="Survive 100 Days In Circle, Win $500,000"
        type="text/html"
        width="100%"
      />
      <div
        class="p-8"
      >
        <h2
          class="text-xl font-bold"
        >
          Survive 100 Days In Circle, Win $500,000
        </h2>
        <pre
          class="whitespace-pre-wrap"
        >
          sign
        </pre>
      </div>
    </article>
    <section
      class="basis-2/6"
    />
  </section>
</DocumentFragment>
`;
